### PR TITLE
Nadav/make safe print safe

### PR DESCRIPTION
**Pull request by nadav-ab**

### CODE REVIEW

File: packages/mostly-equal/src/mostly-equal-chai-plugin.ts [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+  maxDepth?: number;
+}
+
 export const mostlyEqlChaiPlugin: Chai.ChaiPlugin = (c) => {
-  c.Assertion.addMethod('mostlyEqual', function (this, expected) {
-    const res = checkExpectValues(errorString(expected, this._obj, 0, [], new Map(), new Set()));
+  c.Assertion.addMethod('mostlyEqual', function (this, expected, options) {
+    const maxDepth = (options as MostlyEqualOptions)?.maxDepth || 10;
+    const res = checkExpectValues(errorString(expected, this._obj, maxDepth, 0, [], new Map(), new Set()));
     let error = false;
```
Comment (AviVahl) [inline packages/mostly-equal/src/mostly-equal-chai-plugin.ts:12]: Use ?? So we can pass 0 to disable depth limit

File: packages/mostly-equal/src/mostly-equal-chai-plugin.ts [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 
+export interface MostlyEqualOptions {
+  maxDepth?: number;
+}
+
 export const mostlyEqlChaiPlugin: Chai.ChaiPlugin = (c) => {
-  c.Assertion.addMethod('mostlyEqual', function (this, expected) {
-    const res = checkExpectValues(errorString(expected, this._obj, 0, [], new Map(), new Set()));
+  c.Assertion.addMethod('mostlyEqual', function (this, expected, options) {
+    const maxDepth = (options as MostlyEqualOptions)?.maxDepth || 10;
```
Comment (AviVahl) [inline packages/mostly-equal/src/mostly-equal-chai-plugin.ts:10]: Use ?? So we can pass 0 to disable depth limit

File: packages/mostly-equal/mostly-equal-chai.d.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
+interface MostlyEqualOptions {
```
Comment (AviVahl) [inline packages/mostly-equal/mostly-equal-chai.d.ts:1]: put  it under the Chai namespace

File: packages/mostly-equal/src/mostly-equal.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@
 /* eslint-disable @typescript-eslint/no-unsafe-member-access */
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+import { isGetter } from '.';
```
Comment (AviVahl) [inline packages/mostly-equal/src/mostly-equal.ts:7]: circular

Comment (nadav-ab) [reply packages/mostly-equal/src/mostly-equal.ts:7]: done

**Pull request merged**